packages/openapi/src/openapi-request-builder.ts

```diff
@@ -164,23 +164,26 @@ export class OpenApiRequestBuilder<ResponseT = any> {
 
   private getPath(): string {
     const pathParameters = this.parameters?.pathParameters || {};
+    let path = this.pathPattern;
 
-    const pathParts = this.pathPattern.split('/');
-    return pathParts
-      .map(part => {
-        if (OpenApiRequestBuilder.isPlaceholder(part)) {
-          const paramName = part.slice(1, -1);
-          const paramValue = pathParameters[paramName];
-          if (!paramValue) {
-            throw new Error(
-              `Cannot execute request, no path parameter provided for '${paramName}'.`
-            );
-          }
-          return encodeURIComponent(paramValue);
-        }
-        return part;
-      })
-      .join('/');
+    // Get the innermost curly bracket pairs with non-empty and legal content as placeholders.
+    const placeholders = path.match(/{[^/?#{}]+}/g);
+    if (!placeholders) {
+      return path;
+    }
```

Comment:
You can shorten this if you just do the following:
```suggestion
    const placeholders = path.match(/{[^/?#{}]+}/g) || [];
```